index.js

```diff
@@ -847,6 +955,303 @@ function flip(message, side, coinage) {
+                ledger[id].meatCoin += refund;
+                message.author.send('Has been cancelled. You have been credited ' + refund + ' meatCoin\n');
+            }
+            else{
+                var postingData = market[1][id];
+                var xactFee = (postFee) * postingData.quantity;
+                var difference = postingData.quantity - amount;
+                var refund = difference - xactFee;
+                if(refund >= 0){
+                    ledger[id].meatCoin += refund;
+                    message.author.send('Your posting:\n' + id.padStart(18,'0') + '\tS ' + String(postingData.quantity).padStart(10,' ') + '\t@' + String(postingData.price).padStart(8,' ') + '\n');
+                    postingData.price = price;
+                    postingData.quantity = coinage;
+                    response += ', your posting has been updated';
+                    message.author.send('Has been updated to:\n' + id.padStart(18,'0') + '\tS ' + String(postingData.quantity).padStart(10,' ') + '\t@' + String(postingData.price).padStart(8,' ') + '\n');
+                    message.author.send('For a refund of ' + Math.abs(refund).toFixed(2) + ' meatCoin');
+                }
+                else{
+                    //need to pay more check if you have enough
+                    if(ledger[id].meatCoin < Math.abs(refund)){
+                        response += ', you dont have enough money to alter your posting, you need ' + Math.abs(meatCoin) + ' meatCoin.';
+                        message.channel.send(response);
+                        return;
+                    }
+                    //take the extra
+                    else{
+                        ledger[id].meatCoin += refund;
+                        message.author.send('Your posting:\n' + id.padStart(18,'0') + '\tB ' + String(postingData.quantity).padStart(10,' ') + '\t@' + String(postingData.price).padStart(8,' ') + '\n');
+                        postingData.price = price;
+                        postingData.quantity = coinage;
+                        response += ', your posting has been updated';
+                        message.author.send('Has been updated to:\n' + id.padStart(18,'0') + '\tB ' + String(postingData.quantity).padStart(10,' ') + '\t@' + String(postingData.price).padStart(8,' ') + '\n');
+                        message.author.send('For a price of ' + Math.abs(refund).toFixed(2) + ' meatCoin');
+                    }
+                }
+            }
+        }
+        else{
+            response += ', you have no postings!';
+        }
+
+    }
+    else{
+        response += ', ' + command + ' is not a valid command. ya dingus!';
+    }
+
+    message.channel.send(response);
+    processMarket();
+}
+
+function processMarket(){
+    var buyers = market[0];
+    var sellers = market[1];
+    //iterate through buyers
+    if (Object.keys(buyers).length > 0) {
+        Object.keys(buyers).forEach(function(Bkey) {
```

Comment:
Are we doing list iterations here? Can we perhaps use a hash map instead? To find matching buy/sells.